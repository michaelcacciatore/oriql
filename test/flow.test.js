const compileFlow = require('../compile/flow');
const compileTypescript = require('../compile/typescript');

const schema = require('./schema/schema');

describe('Flow', () => {
  it('should compile correctly', () => {
    const compiled = compileFlow(compileTypescript([schema]));

    expect(compiled).toMatchSnapshot();
  });
});
