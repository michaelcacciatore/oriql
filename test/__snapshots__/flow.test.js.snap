// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flow should compile correctly 1`] = `
"export interface PropTypes$testQuery {
  foo?: string | null | void;
  lol?: Array<string> | null | void;
  bar?: string;
  test: Array<{
    michael?: string,
    brielle?: string | null | void,
    robert: {
      itworks?: string | null | void,
      theresa?: string | null | void,
      ...
    } | null | void,
    ...
  }> | null | void;
  mom: {
    michael?: string,
    brielle?: string | null | void,
    robert: {
      itworks?: string | null | void,
      theresa?: string | null | void,
      ...
    } | null | void,
    ...
  } | null | void;
  nestedNoName: {
    hasObject?: string | null | void,
    double: {
      isDouble?: string | null | void,
      ...
    } | null | void,
    ...
  } | null | void;
  firstName?: string | null | void;
  employees: Array<{
    id?: string | null | void,
    employee_name?: string | null | void,
    employee_salary?: string | null | void,
    employee_age?: string | null | void,
    profile_image?: string | null | void,
    ...
  }> | null | void;
}
"
`;
